File: sorts/bubble_sort.js
/**
 * Bubble Sort implementation in JavaScript
 * 
 * @param {Array} collection - The array to sort
 * @returns {Array} - The sorted array
 */
function bubbleSort(collection) {
    let len = collection.length;
    let swapped;

    for (let i = 0; i < len; i++) {
        swapped = false;
        for (let j = 0; j < len - i - 1; j++) {
            if (collection[j] > collection[j + 1]) {
                // Swap elements
                let temp = collection[j];
                collection[j] = collection[j + 1];
                collection[j + 1] = temp;
                swapped = true;
            }
        }
        // If no two elements were swapped by inner loop, then break
        if (!swapped) break;
    }
    return collection;
}

// Example usage
const arr = [64, 34, 25, 12, 22, 11, 90];
console.log("Unsorted array:", arr.slice()); // Copy to display original
console.log("Sorted array:", bubbleSort(arr));

File: sorts/bubble_sort.rs
/**
 * Bubble Sort implementation in Rust
 */

pub fn bubble_sort<T: Ord + Copy>(arr: &mut [T]) {
    let n = arr.len();
    if n == 0 {
        return;
    }

    for i in 0..n {
        let mut swapped = false;
        for j in 0..n - i - 1 {
            if arr[j] > arr[j + 1] {
                arr.swap(j, j + 1);
                swapped = true;
            }
        }
        if !swapped {
            break;
        }
    }
}

fn main() {
    let mut arr = vec![64, 34, 25, 12, 22, 11, 90];
    println!("Unsorted array: {:?}", arr);
    
    bubble_sort(&mut arr);
    
    println!("Sorted array: {:?}", arr);
}
